cypress/integration/single-beacon-owner/i-can-remove-a-use.spec.ts

```diff
@@ -1,67 +1,73 @@
 import { Purpose } from "../../../src/lib/registration/types";
 import { givenIHaveEnteredMyBeaconDetails } from "../common/i-can-enter-beacon-information.spec";
 import {
-  givenIHaveEnteredMyAviationUse,
+  andIHaveEnteredMyAviationUse,
   iCanSeeMyAviationUse,
 } from "../common/i-can-enter-use-information/aviation.spec";
-import { andIHaveAnotherUse } from "../common/i-can-enter-use-information/generic.spec";
+import {
+  andIHaveAnotherUse,
+  whenIHaveAnotherUse,
+} from "../common/i-can-enter-use-information/generic.spec";
 import {
   givenIHaveEnteredMyLandUse,
-  iCanEditMyAdditionalLandUseMoreDetails,
   iCanSeeMyLandUse,
 } from "../common/i-can-enter-use-information/land.spec";
 import {
   givenIHaveEnteredMyMaritimeUse,
   iCanSeeMyMaritimeUse,
 } from "../common/i-can-enter-use-information/maritime.spec";
 import {
-  whenIClickBack,
+  andIClickTheButtonContaining,
   whenIClickTheButtonContaining,
 } from "../common/selectors-and-assertions.spec";
 
 describe("As a beacon owner with several uses", () => {
   it("I can safely remove a use from my draft registration", () => {
-    givenIHaveThreeUses();
+    givenIHaveEnteredMyBeaconDetails();
+    givenIHaveEnteredMyLandUse();
+    andIHaveAnotherUse();
+    givenIHaveEnteredMyMaritimeUse(Purpose.PLEASURE);
+    thereAreNUses(2);
+
     whenIGoToDeleteMyMainUse();
-    thenIAmPromptedToConfirmDeletionOfMyMainUse();
+    thenIAmPromptedToConfirmDeletionOfMyLandUse();
 
     whenIClickTheButtonContaining("Cancel");
-    iCanSeeMyThreeUses();
+    thereAreNUses(2);
+    iCanSeeMyLandUse();
+    iCanSeeMyMaritimeUse(Purpose.PLEASURE);
 
     whenIGoToDeleteMySecondUse();
-    iAmPromptedToConfirmDeletionOfMySecondUse();
-
+    iAmPromptedToConfirmDeletionOfMyMaritimeMotorPleasureUse();
     whenIClickTheButtonContaining("Yes");
-    iCanSeeMyMainUse();
-    iCannotSeeWhatWasMySecondUseBecauseItIsDeleted();
-    myThirdUseIsNowMySecondUse();
+    thereAreNUses(1);
+    iCanSeeMyLandUse();
+    iCannotSeeMyMaritimePleasureUseBecauseItIsDeleted();
 
-    whenIClickBack();
-    iAmEditingWhatIsNowMySecondUse();
-  });
-});
+    whenIHaveAnotherUse();
+    andIHaveEnteredMyAviationUse(Purpose.COMMERCIAL);
+    thereAreNUses(2);
+    iCanSeeMyLandUse();
+    iCanSeeMyAviationUse(Purpose.COMMERCIAL);
 
-const iCanSeeMyMainUse = () => iCanSeeMyLandUse();
+    whenIGoToDeleteMyMainUse();
+    andIClickTheButtonContaining("Yes");
+    thereAreNUses(1);
+    myAviationCommercialUseIsNowMyMainUse();
 
-const myThirdUseIsNowMySecondUse = () =>
-  cy.get("h2").contains(/(?=.*second use)(?=.*aviation)(?=.*commercial)/i);
+    whenIGoToDeleteMyMainUse();
+    whenIClickTheButtonContaining("Yes");
+    thereAreNUses(0);
```

Comment:
Two changes here:
1.  Add Cypress test coverage for the bug.
2. Make the test a _bit_ easier to grok by adding things like `thereAreNUses(1)`.  I was finding it difficult to work out how many and what uses there should be at a given stage in the test.  Hopefully this clears things up!